src/aosm/azext_aosm/custom.py

```diff
@@ -0,0 +1,527 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+import os
+import shutil
+from dataclasses import asdict
+from pathlib import Path
+from typing import Optional
+
+from azure.cli.core.azclierror import (
+    CLIInternalError,
+    InvalidArgumentValueError,
+    UnclassifiedUserFault,
+)
+from azure.cli.core.commands import AzCliCommand
+from azure.core import exceptions as azure_exceptions
+from knack.log import get_logger
+
+from azext_aosm._client_factory import cf_features, cf_resources
+from azext_aosm._configuration import (
+    CNFConfiguration,
+    Configuration,
+    NFConfiguration,
+    NSConfiguration,
+    VNFConfiguration,
+    get_configuration,
+)
+from azext_aosm.delete.delete import ResourceDeleter
+from azext_aosm.deploy.deploy_with_arm import DeployerViaArm
+from azext_aosm.generate_nfd.cnf_nfd_generator import CnfNfdGenerator
+from azext_aosm.generate_nfd.nfd_generator_base import NFDGenerator
+from azext_aosm.generate_nfd.vnf_nfd_generator import VnfNfdGenerator
+from azext_aosm.generate_nsd.nsd_generator import NSDGenerator
+from azext_aosm.util.constants import (
+    AOSM_FEATURE_NAMESPACE,
+    AOSM_REQUIRED_FEATURES,
+    CNF,
+    NSD,
+    VNF,
+    DeployableResourceTypes,
+    SkipSteps,
+)
+from azext_aosm.util.management_clients import ApiClients
+from azext_aosm.vendored_sdks import HybridNetworkManagementClient
+
+logger = get_logger(__name__)
+
+
+def build_definition(
+    definition_type: str,
+    config_file: str,
+    order_params: bool = False,
+    interactive: bool = False,
+    force: bool = False,
+):
+    """
+    Build a definition.
+
+    :param definition_type: VNF or CNF
+    :param config_file: path to the file
+    :param definition_type: VNF, CNF
+    :param interactive - whether to prompt for input when creating deploy parameters
+                         mapping files
+    :param force: force the build even if the design has already been built
+    """
```

Comment:
just a reminder, the parameter of `order_params` is missing from the function documentation.

Reply:
Fixed